Classes/Finishers/EmailFinisher.php

```diff
@@ -213,19 +213,22 @@ protected function addAttachments(SwiftMailerMessage $mail)
                 }
             }
         }
-        foreach ($this->parseOption('attachments') as $attachmentConfiguration) {
-            if (isset($attachmentConfiguration['resource'])) {
-                $mail->attach(\Swift_Attachment::fromPath($attachmentConfiguration['resource']));
-                continue;
-            }
-            if (!isset($attachmentConfiguration['formElement'])) {
-                throw new FinisherException('The "attachments" options need to specify a "resource" path or a "formElement" containing the resource to attach', 1503396636);
-            }
-            $resource = ObjectAccess::getPropertyPath($formValues, $attachmentConfiguration['formElement']);
-            if (!$resource instanceof PersistentResource) {
-                continue;
+
+        if ($this->parseOption('attachments')) {
```

Comment:
we could omit parsing the option twice by storing it in a variable. Besides this would still fail with `attachments` being "1" for example:
```php
$attachments = $this->parseOption('attachments');
if (is_array($attachments)) {
 // ...
```

But I wonder why we need this at all. In line 72 the option defaults to an empty array. If this is not an array it's probably an error in the options configuration

Reply:
@bwaidelich i ran into this in a finisher that extended the classic mail finisher and that stopped working after an update. So i think double checking makes still sense.

Reply:
Could you investigate where the value is overridden and what value it has afterwards?

Reply:
The inherited mail-finisher had his own defaultOptions. So there the attachments were obviously missing.

BTW: The double parsing is removed now.